src/main/java/org/broadinstitute/hellbender/tools/funcotator/Funcotator.java

```diff
@@ -452,31 +459,23 @@ private void enqueueAndHandleVariant(final VariantContext variant, final Referen
         // Get our feature inputs:
         final Map<String, List<Feature>> featureSourceMap = new HashMap<>();
 
-        // Create a variant context for annotation that has a new contig based on whether we need to overwrite the
-        // contig names in the next section.
-        VariantContext variantContextFixedContigForDataSources = variant;
-
         // Check to see if we need to query with a different reference convention (i.e. "chr1" vs "1").
-        if (allowHg19ContigNamesWithB37 && inputReferenceIsB37) {
-            final VariantContextBuilder variantContextBuilderForFixedContigForDataSources = new VariantContextBuilder(variant);
+        if (allowHg19ContigNamesWithB37 && (inputReferenceIsB37 || allowHg19ContigNamesWithB37Lenient)) {
 
             // Construct a new contig and new interval with no "chr" in front of it:
             final String hg19Contig = FuncotatorUtils.convertB37ContigToHg19Contig( variant.getContig() );
             final SimpleInterval hg19Interval = new SimpleInterval(hg19Contig, variant.getStart(), variant.getEnd());
 
-            variantContextBuilderForFixedContigForDataSources.chr(hg19Contig);
-
             // Get our features for the new interval:
             for ( final FeatureInput<? extends Feature> featureInput : manualLocatableFeatureInputs ) {
                 @SuppressWarnings("unchecked")
                 final List<Feature> featureList = (List<Feature>)featureContext.getValues(featureInput, hg19Interval);
-
-                // TODO: This is a little sloppy, since it checks every datasource twice.  Once for hg19 contig names and once for b37 contig names.  See https://github.com/broadinstitute/gatk/issues/4798
-                featureList.addAll(featureContext.getValues(featureInput));
+                if (featureList.size() == 0) {
```

Comment:
In certain cases, we could get multiple hits that were identical.  This caused issues downstream, since de-duping became necessary.

I added a comment for clarification.  

Done.